### PR TITLE
ENH: add 'reflect' extension mode

### PR DESCRIPTION
**Pull request by grlee77**
This `reflect` extension mode was requested in #187.   It is also supported in recent Matlab versions.  It is nearly the same as edge mode `symmetric` but doesn't duplicate the value at the very edge upon reflection.

I have added a couple of tests to verify the expected behavior.  One of the new tests also confirms proper behavior when the DWT filter length exceeds the signal length for various edge modes.

A table comparing the mode naming conventions of `PyWavelets` as compared to Matlab's `dwtmode` and `numpy.pad` was added to the documentation.

~~I don't think it is a blocker, but ideally one would also update the Matlab comparisons to include this edge mode too (it is called `symw` in Matlab), but I am not currently working from a PC with Matlab available.~~
**edit:**  Matlab tests now updated

### CODE REVIEW

**Comment by grlee77** [comment]
There are also two anti-symmetric edge modes in Matlab's `dwtmode` that I believe correspond to 'symmetric' and 'reflect', but with the values negated upon reflection.  I did not add those here, but they would be pretty straightforward as well if there is a use for them.

**Comment by grlee77** [comment]
I updated the precomputed Matlab results.  For simplicity, I just replaced the previous `.npz` file with an updated one that includes the new mode  (file size increased from ~2MB to ~2.4MB). 

The alternative would be to leave the original binary untouched and add a supplementary one for the  `reflect` case.

**Comment by rgommers** [comment]
updating sounds fine. if the tests still pass that gives enough confidence I'd say. and we can always go back to figure out differences at the Nth decimal if really needed.

**Comment by grlee77** [comment]
Any other comments for this one?  If not, will merge soon.

File: pywt/tests/test_modes.py [hunk 1]
```diff
@@ -61,6 +65,31 @@ def test_dwt_idwt_allmodes():
...
 
+def test_dwt_short_input_allmodes():
+    # some test cases where the input is shorter than the DWT filter
+    x = [1, 2, 3]
+    wavelet = 'db2'
+    # manually pad each end by the filter size (4 for 'db2' used here)
+    padded_x = {'zero': [0, 0, 0, 0, 1, 2, 3, 0, 0, 0, 0],
+                'constant': [1, 1, 1, 1, 1, 2, 3, 3, 3, 3, 3],
+                'symmetric': [3, 3, 2, 1, 1, 2, 3, 3, 2, 1, 1],
+                # 'reflect': [1, 2, 3, 2, 1, 2, 3, 2, 1, 2, 3],
```
Comment (rgommers) [inline pywt/tests/test_modes.py]: why is ``'reflect'`` commented out here?

**Comment by rgommers** [comment]
Looks good to me, nice enhancement.

File: pywt/tests/test_modes.py (hunk 1 above)
Comment (grlee77) [inline pywt/tests/test_modes.py]: should not have been commented.  The test was failing in that case and I need to fix it.

Basically, these are tests for when the input signal length is less than the wavelet filter length.  That case (likely pretty rare in practice) is not currently covered by the Matlab tests (they start with signals of size w.dec_len)

**Pull request merged**